### PR TITLE
Search query work done - TP-2

### PR DESCRIPTION
**Pull request by ahsankhan1911**
done with smart query search

### CODE REVIEW

File: api/airports/airports.controller.js [hunk 1]
```diff
@@ -2,26 +2,47 @@ const mongoose = require('mongoose');
...
 
-exports.search =(req, res, next) => {
+exports.search = (req, res, next) => {
 
+    let search = req.query.data;
 
-}
+    if (search.length <= 2) {
+        next(boom.badImplementation("Please provide atleast 3 letters to search"));
+    } else {
```
Comment (khurrumqureshi) [inline api/airports/airports.controller.js]: Remove else. Instead

if (search.length <= 2) {
     return  next(boom.badImplementation("Please provide atleast 3 letters to search"));
 }

File: api/airports/airports.controller.js [hunk 2]
```diff
@@ -2,26 +2,47 @@ const mongoose = require('mongoose');
...
 
+        Airports.find({name: {$regex: search,$options: 'i'}})
 
+            .then((doc) => {
+                if (doc.length === 0) {
 
+                    next(boom.badImplementation("no such airport with keyword " + search + " found"));
+                }
```
Comment (khurrumqureshi) [inline api/airports/airports.controller.js]: Remove else instead use this

if (doc.length === 0) {
  return next(boom.badImplementation("no such airport with keyword " + search + " found"));
 }

**Comment by ahsankhan1911** [comment]
iata added for search

File: api/airports/airports.controller.js [hunk 3]
```diff
@@ -2,26 +2,48 @@ const mongoose = require('mongoose');
...
 
+    let search = req.query.data;
+    let search2 = req.query.data2;
 
-}
+    if (search.length <= 2 || search2.length <= 2) {
+       return next(boom.badImplementation("Please provide atleast 3 letters to search"));
+    } 
 
+        Airports.find({name: {$regex: search,$options: 'i'}, iata:{$regex: search2,$options: 'i'} })
```
Comment (khurrumqureshi) [inline api/airports/airports.controller.js]: we don't need to separate search parameters. User will pass only one search parameter and based on that we need to search against name and iata.

Comment (ahsankhan1911) [reply api/airports/airports.controller.js]: OK

Comment (ahsankhan1911) [reply api/airports/airports.controller.js]: I have fixed the code you can review it..
https://github.com/khurrumqureshi/test-portal/tree/TP-2

**Comment by khurrumqureshi** [comment]
@ahsankhan1911 resolve conflicts.

**Comment by ahsankhan1911** [comment]
I have the conflict.

**Comment by ahsankhan1911** [comment]
I have resolved***

**Pull request merged**